addons/web/static/src/views/debug_items.js

```diff
@@ -217,26 +213,31 @@ class SetDefaultDialog extends Component {
         return this.fieldNamesInView
             .filter((fieldName) => !this.fieldNamesBlackList.includes(fieldName))
             .map((fieldName) => {
-                const modifierData = this.modifierDatas[fieldName];
-                let invisibleOrReadOnly;
-                if (modifierData) {
-                    const evalContext = this.props.component.model.root.evalContext;
-                    invisibleOrReadOnly =
-                        evalDomain(modifierData.invisible, evalContext) ||
-                        evalDomain(modifierData.readonly, evalContext);
-                }
                 const fieldInfo = this.fields[fieldName];
                 const valueDisplayed = this.display(fieldInfo, this.fieldsValues[fieldName]);
                 const value = valueDisplayed[0];
                 const displayed = valueDisplayed[1];
                 // ignore fields which are empty, invisible, readonly, o2m or m2m
                 if (
                     !value ||
-                    invisibleOrReadOnly ||
+                    evalDomain(
+                        this.activeFields[fieldName].invisible,
+                        this.props.record.evalContext
```

Comment:
maybe extract something in a variable (the evalContext or/and the activeField) s.t. this is on a single line